gdal/frmts/netcdf/netcdflayer.cpp

```diff
@@ -118,20 +118,41 @@ static void netCDFWriteAttributesFromConf(
         }
         else if( EQUAL(oAtt.m_osType, "string") )
         {
-            status = nc_put_att_text(cdfid, varid, oAtt.m_osName,
-                                     oAtt.m_osValue.size(), oAtt.m_osValue);
+            if(m_bLegacyCreateMode)
+            {
+                status = nc_put_att_text(cdfid, varid, oAtt.m_osName,
+                                         oAtt.m_osValue.size(), oAtt.m_osValue);
+            }
+            else
+            {
+                m_poDS->vcdf.nc_put_vatt_text(varid, oAtt.m_osName, oAtt.m_osValue);
+            }
         }
         else if( EQUAL(oAtt.m_osType, "integer") )
         {
             int nVal = atoi(oAtt.m_osValue);
-            status =
-                nc_put_att_int(cdfid, varid, oAtt.m_osName, NC_INT, 1, &nVal);
+            if(m_bLegacyCreateMode)
+            {
+                status =
+                    nc_put_att_int(cdfid, varid, oAtt.m_osName, NC_INT, 1, &nVal);
+            }
+            else
+            {
+                m_poDS->vcdf.nc_put_vatt_int(varid, oAtt.m_osName, &nVal);
+            }
         }
         else if( EQUAL(oAtt.m_osType, "double") )
         {
             double dfVal = CPLAtof(oAtt.m_osValue);
-            status = nc_put_att_double(cdfid, varid, oAtt.m_osName,
+            if(m_bLegacyCreateMode)
```

Comment:
see above remarks. could be merged in a single code path with a passthrough vcdf implementation